uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -25,6 +25,14 @@ public static function create($params, $validate = true) {
     $hook = empty($params['id']) ? 'create' : 'edit';
 
     CRM_Utils_Hook::pre($hook, $entityName, CRM_Utils_Array::value('id', $params), $params);
+
+    //For single day leave requests
+    $hasFromDateAndFromDateType = !empty($params['from_date']) && !empty($params['from_date_type']);
+    $doesNotHaveToDateAndToDateType = empty($params['to_date']) && empty($params['to_date_type']);
+    if ($hasFromDateAndFromDateType && $doesNotHaveToDateAndToDateType){
+      $params['to_date'] = $params['from_date'];
+      $params['to_date_type'] = $params['from_date_type'];
+    }
```

Comment:
I just realized that it doesn't make sense to make to_date required and don't do the same with to_date_type. Can you please make to_date_type required as well? Then I think you can remove these 6 lines here